device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -10,7 +10,7 @@
 import lombok.extern.slf4j.Slf4j;
 
 import java.io.IOException;
-import java.util.LinkedList;
+import java.util.ArrayList;
 import java.util.List;
```

Comment:
Why array list?

Reply:
David recommended this change. Arraylist performs better at get/set operations, and linkedlist performs better at add/remove operations. Since we are getting more often then adding, arraylist is the better choice